tests/components/zha/test_config_flow.py

```diff
@@ -154,104 +154,180 @@ def com_port(device="/dev/ttyUSB1234") -> ListPortInfo:
+        (
+            # TubesZB, newer devices
+            "TubeZB",
+            "tubeszb-cc2652-poe",
+            RadioType.znp,
+            zeroconf.ZeroconfServiceInfo(
+                ip_address=ip_address("192.168.1.200"),
+                ip_addresses=[ip_address("192.168.1.200")],
+                hostname="tubeszb-cc2652-poe.local.",
+                name="tubeszb-cc2652-poe._tubeszb._tcp.local.",
+                port=6638,
+                properties={
+                    "name": "TubeZB",
+                    "radio_type": "znp",
+                    "version": "1.0",
+                    "baud_rate": "115200",
+                    "data_flow_control": "software",
+                },
+                type="_tubeszb._tcp.local.",
+            ),
+        ),
+        (
+            # Expected format for all new devices
+            "Some Zigbee Gateway (12345)",
+            "aabbccddeeff",
+            RadioType.znp,
+            zeroconf.ZeroconfServiceInfo(
+                ip_address=ip_address("192.168.1.200"),
+                ip_addresses=[ip_address("192.168.1.200")],
+                hostname="some-zigbee-gateway-12345.local.",
+                name="Some Zigbee Gateway (12345)._zigbee-coordinator._tcp.local.",
+                port=6638,
+                properties={"radio_type": "znp", "serial_number": "aabbccddeeff"},
+                type="_zigbee-coordinator._tcp.local.",
+            ),
+        ),
+    ],
+)
 @patch("homeassistant.components.zha.async_setup_entry", AsyncMock(return_value=True))
 @patch(f"zigpy_znp.{PROBE_FUNCTION_PATH}", AsyncMock(return_value=True))
-async def test_zeroconf_discovery_znp(hass: HomeAssistant) -> None:
+@patch(f"bellows.{PROBE_FUNCTION_PATH}", AsyncMock(return_value=True))
+async def test_zeroconf_discovery(
+    entry_name: str,
+    unique_id: str,
+    radio_type: RadioType,
+    service_info: zeroconf.ZeroconfServiceInfo,
+    hass: HomeAssistant,
+) -> None:
     """Test zeroconf flow -- radio detected."""
-    service_info = zeroconf.ZeroconfServiceInfo(
-        ip_address=ip_address("192.168.1.200"),
-        ip_addresses=[ip_address("192.168.1.200")],
-        hostname="tube._tube_zb_gw._tcp.local.",
-        name="tube",
-        port=6053,
-        properties={"name": "tube_123456"},
-        type="mock_type",
-    )
-    flow = await hass.config_entries.flow.async_init(
+    result = await hass.config_entries.flow.async_init(
         DOMAIN, context={"source": SOURCE_ZEROCONF}, data=service_info
     )
-    assert flow["step_id"] == "confirm"
-
-    # Confirm discovery
-    result1 = await hass.config_entries.flow.async_configure(
-        flow["flow_id"], user_input={}
-    )
-    assert result1["step_id"] == "manual_port_config"
+    assert result["step_id"] == "confirm"
 
     # Confirm port settings
-    result2 = await hass.config_entries.flow.async_configure(
-        result1["flow_id"], user_input={}
+    result = await hass.config_entries.flow.async_configure(
```

Comment:
could we have different `result` vars as before, to avoid reusing the same var for multiple things?
maybe with a more descriptive name like `result_init`, `result_config`, `result_formation`, etc?